## Medidas de posição

In [1]:
# carrega dependencias
import numpy as np
import pandas as pd

In [2]:
# monta drive
from google.colab import drive
drive.mount('/content/drive')

Mounted at /content/drive


In [3]:
# carrega os dados
enem_sp = pd.read_csv('/content/drive/MyDrive/UDEMY - Estatística para Análise de Dados/Moreno Marcos/microdados_enem_2019_sp_tratado.csv', sep=',', encoding='iso-8859-1')
enem_sp

NO_MUNICIPIO_RESIDENCIA  IDADE SEXO  TP_ESTADO_CIVIL    RACA  \
0                 Santa Isabel     22    M                1   parda   
1                    São Paulo     17    F                1   parda   
2        São Bernardo do Campo     19    M                1   parda   
3                   Embu-Guaçu     17    F                1   parda   
4                    Guarulhos     23    F                1  branca   
...                        ...    ...  ...              ...     ...   
499935               São Paulo     18    F                1   parda   
499936               São Paulo     17    F                1   preta   
499937               São Paulo     17    F                1   parda   
499938               São Paulo     17    M                1   parda   
499939                  Araras     18    F                1  branca   

        TP_NACIONALIDADE NO_MUNICIPIO_NASCIMENTO SG_UF_NASCIMENTO  \
0                      1    Santa Izabel do Pará               PA   
1                      1                   Belém               PA   
2                      1   São Bernardo do Campo               SP   
3                      1             Itacoatiara               AM   
4                      1               São Paulo               SP   
...                  ...                     ...              ...   
499935                 1               São Paulo               SP   
499936                 1               São Paulo               SP   
499937                 1               São Paulo               SP   
499938                 1               São Paulo               SP   
499939                 1                  Araras               SP   

        TP_ST_CONCLUSAO  TP_ANO_CONCLUIU  ... NOTA_MT  TP_LINGUA  \
0                     1                4  ...   600.2     Inglês   
1                     2                0  ...   731.8     Inglês   
2                     2                0  ...   393.2   Espanhol   
3                     2                0  ...   419.2     Inglês   
4                     1                5  ...   796.1     Inglês   
...                 ...              ...  ...     ...        ...   
499935                1                1  ...   479.1   Espanhol   
499936                2                0  ...    43.2   Espanhol   
499937                2                0  ...   487.3     Inglês   
499938                2                0  ...   401.1     Inglês   
499939                2                0  ...   559.5     Inglês   

        TP_STATUS_REDACAO  COMP1  COMP2  COMP3  COMP4  COMP5  NOTA_REDACAO  \
0                     1.0  160.0  200.0  180.0  200.0  200.0         940.0   
1                     1.0  160.0  120.0  200.0  200.0  200.0         880.0   
2                     1.0   80.0  160.0  120.0  100.0  100.0         560.0   
3                     1.0  120.0  120.0  120.0  120.0   80.0         560.0   
4                     1.0  160.0  200.0  180.0  180.0  160.0         880.0   
...                   ...    ...    ...    ...    ...    ...           ...   
499935                1.0  100.0  120.0  120.0  140.0  100.0         580.0   
499936                1.0   80.0  120.0  140.0  120.0  100.0         560.0   
499937                1.0  120.0   80.0   60.0  120.0    0.0         380.0   
499938                1.0  160.0  120.0  160.0  140.0  160.0         740.0   
499939                1.0  120.0  120.0  120.0  120.0  140.0         620.0   

        INTERNET  
0            Sim  
1            Não  
2            Sim  
3            Sim  
4            Sim  
...          ...  
499935       Sim  
499936       Sim  
499937       Sim  
499938       Não  
499939       Sim  

[499940 rows x 35 columns]

## Medidas de Posição

Selecionamos a nota da matemáticas dos candidatos do ENEM em SP para acharmos a mínima nota obtida.

In [4]:
enem_sp['NOTA_MT'].min()

0.0

In [5]:
enem_sp.NOTA_MT.min()

0.0

Isto sugere que, praticamente teve alunos que não estudaram absolutamente nada para essa prova.

Analogamente, achamos a nota máxima obtida em Matemáticas.

In [6]:
enem_sp['NOTA_MT'].max()

984.7

Isto sugere que teve alunos, que quase atingiram a máxima pontuação.

Agora, vamos a achar os Q1, Q2 e Q3 (quartis).

In [7]:
# Primeiro Quartil
enem_sp['NOTA_MT'].quantile(q=0.25)

427.9

O 25% do total das notas dos alunos concentrou-se no intervalo de 0 (nota minima) até a nota 428 aprox.

In [8]:
# Segundo Quartil (Mediana)
enem_sp['NOTA_MT'].quantile(q=0.5)

511.8

O 50% do total das notas se concentrou no interval de 0 (nota minima) até a nota 512 aprox.

In [10]:
# Terceiro Quartil
enem_sp['NOTA_MT'].quantile(q=0.75)

617.9

O 75% do total de notas se concetra no interval de 0 (nota minima) até a nota 618 aprox.

Podemos também calcular todas as medidas de centralidade, dispersão e posição de uma ves só para a nota de matemática.

In [12]:
enem_sp.NOTA_MT.describe()

count    499940.000000
mean        497.594240
std         183.465175
min           0.000000
25%         427.900000
50%         511.800000
75%         617.900000
max         984.700000
Name: NOTA_MT, dtype: float64

Agora iremos a analisar brevemente a notas da redação com suas componente.

In [14]:
enem_sp[['COMP1', 'COMP2', 'COMP3', 'COMP4', 'COMP5', 'NOTA_REDACAO']].describe()

COMP1          COMP2          COMP3          COMP4  \
count  499940.000000  499940.000000  499940.000000  499940.000000   
mean      125.000760     127.066808     118.760691     124.989359   
std        28.629513      39.846083      36.778882      33.879299   
min         0.000000       0.000000       0.000000       0.000000   
25%       120.000000     120.000000     100.000000     120.000000   
50%       120.000000     120.000000     120.000000     120.000000   
75%       140.000000     140.000000     140.000000     140.000000   
max       200.000000     200.000000     200.000000     200.000000   

               COMP5   NOTA_REDACAO  
count  499940.000000  499940.000000  
mean       97.200704     593.018322  
std        57.787427     170.536724  
min         0.000000       0.000000  
25%        60.000000     500.000000  
50%       100.000000     600.000000  
75%       140.000000     680.000000  
max       200.000000    1000.000000

Em relação aos quartis das componentes:
- O Q1 e Q2 das componentes 1, 2 e 4 ao serem iguais a 120 pontos, isto sugere que aprox. o 25% do total das notas dos alunos é igual a 120 pontos;
- Note-se que para todas as componentes, o Q3 é o mesmo e igual a 140 pontos;
- Dentre todas as Q1 das componentes, a Q1 da COMP5 possui um intervalo de notas menor [0 - 60] que incorpora o 25% das notas do total de alunos.